xcodeproj/internal/linker_input_files.bzl

```diff
@@ -366,6 +366,13 @@ def _process_linkopt(linkopt, *, triple):
         if opt.endswith(".swiftmodule"):
             opt = opt + "/{}.swiftmodule".format(triple)
 
+        # Process paths in the --flag=value format
+        if "=" in opt:
```

Comment:
Since partition needs to check for the separator anyway, I think you can always call that, and then based on a None value or not determine which part to apply the path logic to.